lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java

```diff
@@ -103,6 +103,6 @@ public void destroy()
     @Override
     public TrinoFileSystem create(ConnectorIdentity identity)
     {
-        return new S3FileSystem(client, context);
+        return new S3FileSystem(client, context, null);
```

Comment:
This option is always null. What is the purpose for adding it? Can we wait to add it until there is a real use case?

Reply:
@electrum thanks for the review.
yeah it's always null for `S3FileSystemFactory`. We plan to have a custom implementation of `TrinoFileSystem` factory, similar to `S3FileSystemFactory` where we want to leverage this option. We don't want to have custom impl of `TrinoFileSystem` (similar to `S3FileSystem`).

Reply:
Thanks for the explanation. We shouldn't add this unless it will be used in Trino, as it makes the code base more complicated for no benefit to the Trino community. Someone might later notice that it's unused and remove it.

It's possible that we'll want to add this later if Trino needs it.